db/migrate/20171212151526_add_file_submission_to_question_programming.rb

```diff
@@ -0,0 +1,20 @@
+class AddFileSubmissionToQuestionProgramming < ActiveRecord::Migration[5.0]
+  def up
+    add_column :course_assessment_question_programming, :file_submission, :boolean, default: false, null: false
+
+    Course::Assessment::Question::Programming.find_each do |question|
+      service = Course::Assessment::Question::Programming::ProgrammingPackageService.new(question, nil)
+      meta = service.extract_meta
+      if meta[:data]['submit_as_file']
+        Course::Assessment::Question::Programming.transaction do
```

Comment:
Just putting this out there - not sure whether this will take a long time since `question.save` invokes the activerecord callbacks. The other way is to use `update_attribute`. 

Strictly speaking, [migrations are run within a transaction](http://api.rubyonrails.org/classes/ActiveRecord/Migration.html) so no need to wrap it within another one.